mojo/stdlib/stdlib/python/python_object.mojo

```
@@ -389,15 +389,20 @@ struct PythonObject(
         if not dict_obj_ptr:
             raise Error("internal error: PyDict_New failed")
 
+        # Safety: since we receive a deep copy of the PythonObject values, we
+        # can steal them
         for i in range(len(keys)):
-            var key_obj = keys[i].copy().to_python_object()
-            var val_obj = values[i].copy().to_python_object()
             var result = cpython.PyDict_SetItem(
-                dict_obj_ptr, key_obj._obj_ptr, val_obj._obj_ptr
+                dict_obj_ptr, keys[i]._obj_ptr, values[i]._obj_ptr
             )
             if result != 0:
+                # destroy the incorrectly built dict to avoid a memory leak
+                _ = PythonObject(from_owned=dict_obj_ptr)
```

And for the record, this line is good. However, I'm not sure what the policy is regarding internal errors; perhaps we don't even need to handle the `result != 0` case.

---

Why would we ignore an error? I think this should raise

---

It's kinda a grey area, like how people often ignore the errno of `printf`. Internal error mostly indicates a programming error on our side. I'm not advocating for either side, just IDK.

---

I would rather raise the error than silently fail

---

Noted. I will tag you should we discuss the CPython internal error policy on GH.